devito/dse/backends/advanced.py

```diff
@@ -142,9 +142,8 @@ def _eliminate_inter_stencil_redundancies(self, cluster, template, **kwargs):
 
             # Build a symbolic function for /alias/
             intervals = ispace.intervals
-            shape = tuple(i.symbolic_size for i in indices)
             halo = [(abs(intervals[i].lower), abs(intervals[i].upper)) for i in indices]
-            function = Array(name=template(), shape=shape, dimensions=indices, halo=halo)
+            function = Array(name=template(), dimensions=indices, halo=halo)
```

Comment:
why i shape not necessary anymore?

Reply:
an Array is a purely symbolic thing (e.g., it has associated **no** user data), so it's shape also is purely symbolic. And the symbolic shape can completely be inferred from the ``indices`` and the ``halo``. So shape was meaningless and redundant at the same time here